flash/vision/style_transfer/model.py

```diff
@@ -0,0 +1,210 @@
+        scheduler_kwargs: Optional[Dict[str, Any]] = None,
+        learning_rate: float = 1e-3,
+        serializer: Optional[Union[Serializer, Mapping[str, Serializer]]] = None,
+    ):
+        if isinstance(style_image, str):
+            style_image = read_image(style_image)
+
+        if multi_layer_encoder is None:
+            multi_layer_encoder = self.default_multi_layer_encoder()
+
+        if content_loss is None:
+            content_loss = self.default_content_loss(multi_layer_encoder)
+
+        if style_loss is None:
+            style_loss = self.default_style_loss(multi_layer_encoder)
+
+        self.perceptual_loss = loss.PerceptualLoss(content_loss, style_loss)
+        self.perceptual_loss.set_style_image(style_image)
+
+        self.save_hyperparameters()
+
+        super().__init__(
+            model=model,
+            loss_fn=self.perceptual_loss,
+            optimizer=optimizer,
+            optimizer_kwargs=optimizer_kwargs,
+            scheduler=scheduler,
+            scheduler_kwargs=scheduler_kwargs,
+            learning_rate=learning_rate,
+            serializer=serializer,
+        )
+
+    def default_multi_layer_encoder(self) -> enc.MultiLayerEncoder:
+        return enc.vgg16_multi_layer_encoder()
+
+    def default_content_loss(
+        self, multi_layer_encoder: Optional[enc.MultiLayerEncoder] = None
+    ) -> ops.FeatureReconstructionOperator:
+        if multi_layer_encoder is None:
+            multi_layer_encoder = self.default_multi_layer_encoder()
+        content_layer = "relu2_2"
+        content_encoder = multi_layer_encoder.extract_encoder(content_layer)
+        content_weight = 1e5
+        return ops.FeatureReconstructionOperator(content_encoder, score_weight=content_weight)
+
+    def default_style_loss(
+        self, multi_layer_encoder: Optional[enc.MultiLayerEncoder] = None
+    ) -> ops.MultiLayerEncodingOperator:
+
+        class GramOperator(ops.GramOperator):
+
+            def enc_to_repr(self, enc: torch.Tensor) -> torch.Tensor:
+                repr = super().enc_to_repr(enc)
+                num_channels = repr.size()[1]
+                return repr / num_channels
+
+        if multi_layer_encoder is None:
+            multi_layer_encoder = self.default_multi_layer_encoder()
+
+        style_layers = ("relu1_2", "relu2_2", "relu3_3", "relu4_3")
+        style_weight = 1e10
+        return ops.MultiLayerEncodingOperator(
+            multi_layer_encoder,
+            style_layers,
+            lambda encoder, layer_weight: GramOperator(encoder, score_weight=layer_weight),
+            layer_weights="sum",
+            score_weight=style_weight,
+        )
+
+    def forward(self, content_image: torch.Tensor) -> torch.Tensor:
+        self.perceptual_loss.set_content_image(content_image)
+        return self.model(content_image)
+
+    def validation_step(self, batch: Any, batch_idx: int) -> NoReturn:
```

Comment:
Should also override `train_step` and `predict_step` here to get the `DefaultDataKeys.INPUT` from the batch. Example here: https://github.com/PyTorchLightning/lightning-flash/blob/d5ca5e526ff1cb3bad23a48398c4634b7fa37724/flash/vision/classification/model.py#L110